### PR TITLE
fix: allow relative paths for `--config-file`

### PR DESCRIPTION
**Pull request by sriram-mv**
* PR https://github.com/aws/aws-sam-cli/pull/4044 has caused a
  regression.

#### Which issue(s) does this change fix?
* https://github.com/aws/aws-sam-cli/pull/4044 

#### Why is this change necessary?
* Re allow usage of relative paths for `--config-file`

#### How does it address the issue?
* Only check for the path being a file after resolving it to be absolute.

#### What side effects does this change have?
* N/A

#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [x] Write/update unit tests
- [ ] Write/update integration tests
- [ ] Write/update functional tests if needed
- [x] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: samcli/cli/cli_config_file.py [hunk 1]
```diff
@@ -130,20 +130,20 @@ def configuration_callback(cmd_name, option_name, saved_callback, provider, ctx,
     config_env_name = ctx.params.get("config_env") or DEFAULT_ENV
 
     config_file = ctx.params.get("config_file") or DEFAULT_CONFIG_FILE_NAME
-    if config_file and config_file != DEFAULT_CONFIG_FILE_NAME and not Path(config_file).is_file():
-        error_msg = f"Config file {config_file} does not exist or could not be read"
+    config_dir = getattr(ctx, "samconfig_dir", None) or os.getcwd()
+    # If --config-file is an absolute path, use it, if not, start from config_dir
+    config_file_path = config_file if os.path.isabs(config_file) else os.path.join(config_dir, config_file)
+    if config_file and config_file != DEFAULT_CONFIG_FILE_NAME and not Path(config_file_path).absolute().is_file():
```
Comment (mndeveci) [inline samcli/cli/cli_config_file.py:136]: We didn't have `absolute` call before, why we need it now?

**Review by mndeveci — COMMENTED** [review]
LGTM in general just couple of questions to understand the behavior here.

File: .gitignore [hunk 2]
```diff
@@ -292,6 +292,9 @@ docs/_build/
 # pyenv
 .python-version
 
+# pyright
+pyrightconfig.json
```
[the remark this answers is not in the document]
Comment (sriram-mv) [reply .gitignore:296]: This is my LSP when I'm editing via vim.

File: samcli/cli/cli_config_file.py (hunk 1 above)
Comment (sriram-mv) [reply samcli/cli/cli_config_file.py:136]: if the config file supplied has a relative path, we need to resolve it, which looks like was a regression from the previous PR, which is why I added a test particularly for this.

**Review by mndeveci — APPROVED** [review]

**Review by jfuss — APPROVED** [review]
LGTM

**Pull request merged**